File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/TxDocSequenceDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.md.base.tx.ds.model.TxDocSequenceDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"firstNo", type:"int", useNull:true},
		{name:"id", type:"int", useNull:true},
		{name:"incrementBy", type:"int", useNull:true},
		{name:"lastNo", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"nextNo", type:"int", useNull:true},
		{name:"prefix", type:"string"},
		{name:"suffix", type:"string"},
		{name:"uuid", type:"string"},
		{name:"version", type:"int", useNull:true}
	],
	validations: [
		{field: "name", type: 'presence'}
	]
});

Ext.define("net.nan21.dnet.module.md.base.tx.ds.model.TxDocSequenceDsFilter", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"firstNo", type:"int", useNull:true},
		{name:"firstNo_From",type:"int", useNull:true},
		{name:"firstNo_To",type:"int", useNull:true},
		{name:"id", type:"int", useNull:true},
		{name:"incrementBy", type:"int", useNull:true},
		{name:"incrementBy_From",type:"int", useNull:true},
		{name:"incrementBy_To",type:"int", useNull:true},
		{name:"lastNo", type:"int", useNull:true},
		{name:"lastNo_From",type:"int", useNull:true},
		{name:"lastNo_To",type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"nextNo", type:"int", useNull:true},
		{name:"nextNo_From",type:"int", useNull:true},
		{name:"nextNo_To",type:"int", useNull:true},
		{name:"prefix", type:"string"},
		{name:"suffix", type:"string"},
		{name:"uuid", type:"string"}
	]
});

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/base/tx/ds/model/TxDocSequenceDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.base.tx.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeDs;
import net.nan21.dnet.module.md.base.tx.domain.entity.TxDocSequence;

@Ds(entity = TxDocSequence.class, sort = {@SortField(field = TxDocSequenceDs.f_name)})
public class TxDocSequenceDs extends AbstractTypeDs<TxDocSequence> {

	public static final String f_firstNo = "firstNo";
	public static final String f_lastNo = "lastNo";
	public static final String f_nextNo = "nextNo";
	public static final String f_incrementBy = "incrementBy";
	public static final String f_prefix = "prefix";
	public static final String f_suffix = "suffix";

	@DsField
	private Integer firstNo;

	@DsField
	private Integer lastNo;

	@DsField
	private Integer nextNo;

	@DsField
	private Integer incrementBy;

	@DsField
	private String prefix;

	@DsField
	private String suffix;

	public TxDocSequenceDs() {
		super();
	}

	public TxDocSequenceDs(TxDocSequence e) {
		super(e);
	}

	public Integer getFirstNo() {
		return this.firstNo;
	}

	public void setFirstNo(Integer firstNo) {
		this.firstNo = firstNo;
	}

	public Integer getLastNo() {
		return this.lastNo;
	}

	public void setLastNo(Integer lastNo) {
		this.lastNo = lastNo;
	}

	public Integer getNextNo() {
		return this.nextNo;
	}

	public void setNextNo(Integer nextNo) {
		this.nextNo = nextNo;
	}

	public Integer getIncrementBy() {
		return this.incrementBy;
	}

	public void setIncrementBy(Integer incrementBy) {
		this.incrementBy = incrementBy;
	}

	public String getPrefix() {
		return this.prefix;
	}

	public void setPrefix(String prefix) {
		this.prefix = prefix;
	}

	public String getSuffix() {
		return this.suffix;
	}

	public void setSuffix(String suffix) {
		this.suffix = suffix;
	}
}
